### PR TITLE
feat: improving the Infobox style 

### PR DESCRIPTION
**Pull request by mimoham24**
# Overview
apply new styles to the infobox
## What I've done
applied the new styles provided by the mock up
- infobox position
- infobox new width option
- show infobox title 
- infobox outline 
- infobox use mask option

## How I tested
manual

## Screenshot
Screenshot from 2022-02-13 11-07-20

### CODE REVIEW

**Comment by lavalse** [comment]
@mimoham24  Sorry, I have lost two small points by mistake:
- the black-mask can be clicked and can close ( unselect ) the marker 
- if not set the outline color, it will use the public theme color

when you come, let's discuss for details.

File: src/components/molecules/Visualizer/Infobox/Frame/index.tsx [hunk 1]
```diff
@@ -90,64 +104,110 @@ const InfoBox: React.FC<Props> = ({
...
+                  open={open}
+                />
+                <StyledIcon color={publishedTheme.mainIcon} icon="infobox" size={24} open={open} />
+              </IconWrapper>
+            )}
+            {!open ? null : (
+              <Text size="m" weight="bold" customColor>
+                <TitleText show={showTitle ?? true}>{title || " "}</TitleText>
+              </Text>
+            )}
```
Comment (HideBa) [inline src/components/molecules/Visualizer/Infobox/Frame/index.tsx:145]: Can be simplier

```suggestion
            {open && (
              <Text size="m" weight="bold" customColor>
                <TitleText show={showTitle ?? true}>{title || " "}</TitleText>
              </Text>
            )}
```

File: src/components/molecules/Visualizer/Infobox/Frame/index.tsx [hunk 2]
```diff
@@ -90,64 +104,110 @@ const InfoBox: React.FC<Props> = ({
...
+  outlineColor?: string;
+  outlineWidth?: number;
 }>`
   top: 15%;
-  right: ${({ open }) => (open ? "30px" : "-6px")};
-  max-height: 70%;
-  width: ${({ size, open }) => (open ? (size == "large" ? "624px" : "346px") : "80px")};
+  ${({ open, position }) =>
+    open
+      ? position == "middle"
```
Comment (HideBa) [inline src/components/molecules/Visualizer/Infobox/Frame/index.tsx:195]: Plz use strinct equal
```
//Bad
== 

//OK
===
```

**Review by HideBa — APPROVED** [review]
Mostly good. Plz check my comments. 👍

**Review by HideBa — CHANGES_REQUESTED** [review]
I wrongly approved your PR.

File: src/components/molecules/Visualizer/Infobox/index.tsx [hunk 3]
```diff
@@ -82,9 +83,16 @@ const Infobox: React.FC<Props> = ({
       sceneProperty={sceneProperty}
       title={property?.default?.title || title}
```
Comment (issmail-basel) [inline src/components/molecules/Visualizer/Infobox/index.tsx:84]: You can assign default values here like `title` if needed

**Review by issmail-basel — CHANGES_REQUESTED** [review]
Please fix comments mentioned by @HideBa .
Other than that seems good.

**Comment by lavalse** [comment]
@mimoham24 also, the background mask still has a problem. When the user clicks the mask, what happens is not minimizing the infobox, but to **Unselect** the layer. So can you fix it quickly?

File: src/components/molecules/Visualizer/Infobox/index.tsx (hunk 3 above)
Comment (mimoham24) [reply src/components/molecules/Visualizer/Infobox/index.tsx:84]: didin't get it ... no changes are applied from my side her

**Review by issmail-basel — APPROVED** [review]

**Review by HideBa — APPROVED** [review]

**Pull request merged**